test/Object/fromEntries.js

```diff
@@ -0,0 +1,213 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+function verifyProperties(obj, property, value)
+{
+    const descriptor = Object.getOwnPropertyDescriptor(obj, property);
+    assert.areEqual(value, obj[property], "Object.fromEntries should set correct valued");
+    assert.isTrue(descriptor.enumerable, "Object.fromEntries should create enumerable properties");
+    assert.isTrue(descriptor.configurable, "Object.fromEntries should create configurable properties");
+    assert.isTrue(descriptor.writable, "Object.fromEntries should create writable properties");
+    obj[property] = "other value";
+    assert.areEqual("other value", obj[property], "should actually be able to write to properties created by Object.fromEntries");
+    assert.doesNotThrow(()=>{"use strict"; delete obj[property];}, "deleting properties created by Object.fromEntries should not throw");
+    assert.isUndefined(obj[property], "deleting properties created by Object.fromEntries should succeed");
+}
+
+function verifyObject(expected, actual)
+{
+    for (let i in expected)
```

Comment:
should this include a has own property check on expected, too?

Reply:
Erm - yeah this was a mistake it was meant to do hasProperty checks on expected not on actual (as the verifyProperty bit checks the actuals; it's exactly backwards, will fix.